include/mbgl/util/string.hpp

```diff
@@ -10,7 +10,16 @@ namespace util {
 
 template <class T>
 inline std::string toString(T t) {
-    return std::to_string(t);
+    if (t - (int)t > 0.0 || t + (int)t < 0.0) {
+        std::string str = std::to_string(t);
+        str.erase(str.find_last_not_of('0') + 1, std::string::npos);
+        if (str.back() == '.') {
```

Comment:
Interesting, I definitely did not consider these strings as being localizable — is this a realistic scenario right now?

A possible fix would be to remove any non-number that’s the final character.